service/src/main/java/uk/gov/hmcts/reform/fpl/handlers/CustomAsyncExceptionHandler.java

```diff
@@ -0,0 +1,19 @@
+package uk.gov.hmcts.reform.fpl.handlers;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.aop.interceptor.AsyncUncaughtExceptionHandler;
+
+import java.lang.reflect.Method;
+
+public class CustomAsyncExceptionHandler implements AsyncUncaughtExceptionHandler {
+
+    Logger logger = LoggerFactory.getLogger(getClass());
+
+    @Override
+    public void handleUncaughtException(
+        Throwable throwable, Method method, Object... obj) {
+        logger.error("Exception message - " + throwable.getMessage());
```

Comment:
Please pass exception as well so that stack trace is printed.
```suggestion
        logger.error("Unexpected error occurred during async execution: " + throwable.getMessage(), throwable);
```